packages/beacon-node/src/chain/bls/multithread/jobItem.ts

```diff
@@ -70,6 +73,42 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
       // and not a problem in the near future
       // this is monitored on v1.11.0 https://github.com/ChainSafe/lodestar/pull/5912#issuecomment-1700320307
       const timer = metrics?.blsThreadPool.signatureDeserializationMainThreadDuration.startTimer();
+
+      // adding verification randomness is swig specific. must not attempt with herumi until
+      // @chainsafe/bls is updated to support it with herumi
+      if (job.opts.addVerificationRandomness) {
+        const pkPoint = new swigBindings.blst.P1();
+        const sigPoint = new swigBindings.blst.P2();
+        for (let i = 0; i < job.sets.length; i++) {
+          const randomness = randomBytesNonZero(8);
+          // cast to unknown here because the blst-native version of the bls library extends the
+          // PublicKey from the blst library, but the herumi version does not so the interface does
+          // not show that this is possible
+          const {value} = job.sets[i].publicKey as unknown as blst.PublicKey;
+          let dup: swigBindings.P1;
+          if (typeof (value as swigBindings.P1_Affine).to_jacobian === "function") {
```

Comment:
Is there anything that could be done on `bls` side to remove:
* the blst specific cast
* the `to_jacobian` check
* the `blst.Signature` usage

Might be worth creating an issue in the `bls` repo?

Also, what does that mean if this is executed in `env` not running native `bls` security-wise? (retorical question for now as there is no such env AFAIK)

Reply:
Yes and no.  Short answer is no though.  the mult() functionality is not implemented by the isomorphic bls library because herumi does not support it (see the comment response below).  So we need to step down to the blst lib to get the functionality we need to resolve the issue.  The `to_jacobian` check is implemented in bls sort of (the PublicKey class extends the blst PublicKey class) and parts of that code are actually non-dry and using what is in blst.  

It is possible to use bls to get access to that method but it would take a couple hacks because the constructor of PublicKey is private and we would also need to new up a new object that would immediately get thrown away afterwards so for performance reasons I just skipped that step.  

But for completeness this is what it would look like.  I am open to changing though as this is not something i feel strongly about but hopefully this code will all go away in a few weeks when we get the blst rebuild merged.  This was just an acceleration to get the potential security issue cleared up quickly

```ts
const point = (set.publicKey as unknown as blst.PublicKey).value.dup();  // create dup
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
const pk = new bls.PublicKey(point); // new up bls wrapper objects
pkPoint.add((pk as blst.PublicKey).jacobian.mult(randomness)); // create copy of point with `.jacobian`
```

versus 

```ts
let dup: swigBindings.P1;
if (typeof (value as swigBindings.P1_Affine).to_jacobian === "function") {   // either create dup or fresh point
    dup = (value as swigBindings.P1_Affine).to_jacobian();
} else {
    dup = value.dup() as swigBindings.P1;
}
pkPoint.add(dup.mult(randomness));
```